lib/internal/bootstrap/loaders.js

```diff
@@ -208,6 +211,77 @@
         NativeModule.require;
       fn(this.exports, requireFn, this, process);
 
+      if (config.experimentalModules) {
+        if (this.id === 'events') {
+          this.namespace = [];
+        } else {
+          this.namespace = Object.entries(
+            Object.getOwnPropertyDescriptors(this.exports))
+            .filter(([name, d]) => d.enumerable)
+            .map(([name]) => name);
+        }
+
+        const update = (property, value) => {
+          if (this.reflect !== undefined && property in this.reflect.exports)
+            this.reflect.exports[property].set(value);
+        };
+
+        const methodWrapMap = new WeakMap();
+
+        const proxy = new Proxy(this.exports, {
+          set(target, prop, value) {
+            update(prop, value);
+            target[prop] = value;
+            return true;
+          },
+          defineProperty(target, prop, descriptor) {
+            if (this.namespace.includes(prop)) {
+              const valueDescriptor =
+                Reflect.getOwnPropertyDescriptor(descriptor, 'value');
+              if (valueDescriptor &&
+                  Reflect.defineProperty(target, prop, descriptor)) {
+                update(prop, valueDescriptor.value);
+                return true;
+              }
+            }
+            return Reflect.defineProperty(target, prop, descriptor);
+          },
+          deleteProperty(target, prop) {
+            if (this.namespace.includes(prop))
+              return false;
+            return delete target[prop];
+          },
+          get(target, prop) {
+            const value = Reflect.get(target, prop, target);
+            if (methodWrapMap.has(value))
+              return methodWrapMap.get(value);
+            if (typeof value === 'function') {
+              function wrap(...args) {
+                if (new.target)
+                  return Reflect.construct(value, args, new.target);
+                const thisArg = this === proxy ? target : this;
+                return Reflect.apply(value, thisArg, args);
+              }
+              const { name, length } = value;
+              const string =
+                `function${name ? ` ${name}` : ''}() { [native code] }`;
+              Object.defineProperties(wrap, {
+                name: { value: name },
+                length: { value: length },
+                toString: { value: () => string },
```

Comment:
This won’t be robust against Function.toString.call, fwiw

Reply:
i think thats just left over from the project that i copied this from, i'll remove it